src/boot/efi/shim.c

```diff
@@ -97,3 +97,9 @@ EFI_STATUS shim_load_image(EFI_HANDLE parent, const EFI_DEVICE_PATH *device_path
 
         return ret;
 }
+
+void shim_retain_protocol(void) {
+        uint8_t value = 1;
+
+        (void) efivar_set_raw(MAKE_GUID_PTR(SHIM_LOCK), u"ShimRetainProtocol", &value, sizeof(value), 0);
```

Comment:
Please add comment what the precise effect is, i.e. what ahppens without it and what happens with it. Interaction with shimis a bit cryptic otherwise